### PR TITLE
Add `math.map` to the standard library

### PR DESCRIPTION
**Pull request by MagmaBurnsV**
[Rendered](https://github.com/luau-lang/rfcs/blob/17bfcac8ea5f4bfff9206b0b4bb2fb1fa914a8b4/docs/function-math-map.md)

### CODE REVIEW

File: docs/function-math-map.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+function math.map(x: number, inmin: number, inmax: number, outmin: number, outmax: number): number
+	return outmin + (x - inmin) * (outmax - outmin) / (inmax - inmin)
+end
+```
+
+Just like common Luau implementations, `x` is allowed to be outside the input range. Since clamped mapping is not as widely used and can be easily replicated using `math.clamp`, it would not be supported.
+
+## Drawbacks
+
+This would add a new function to the native library, taking yet another fastcall slot.
```
Comment (vegorov-rbx) [inline docs/function-math-map.md:28]: We do not start with adding every new function to fastcalls, so we will just skip this one, like random\noise.

**Comment by vegorov-rbx** [comment]
Looks alright.
Let's wait for 2 week minimum for other feedback.

**Comment by kalrnlo** [comment]
Could this be merged? Its been a month and theres been no objections.

**Review by vegorov-rbx — APPROVED** [review]

**Comment by vegorov-rbx** [comment]
As a more simple addition, this should be available in next release after the sync.

**Comment by AxisAngles** [comment]
https://gist.github.com/AxisAngles/e3c6ad3f57044480b69ff378477db03a

**Pull request merged**